android/uhabits-android/src/main/java/org/isoron/uhabits/activities/habits/list/ListHabitsMenu.kt

```diff
@@ -33,15 +33,32 @@ class ListHabitsMenu @Inject constructor(
         private val preferences: Preferences,
         private val themeSwitcher: ThemeSwitcher,
         private val behavior: ListHabitsMenuBehavior
-) : BaseMenu(activity) {
+) : BaseMenu(activity), Preferences.Listener {
+
+    private lateinit var topBarMenu: Menu
 
     override fun onCreate(menu: Menu) {
         val nightModeItem = menu.findItem(R.id.actionToggleNightMode)
         val hideArchivedItem = menu.findItem(R.id.actionHideArchived)
         val hideCompletedItem = menu.findItem(R.id.actionHideCompleted)
+
         nightModeItem.isChecked = themeSwitcher.isNightMode
         hideArchivedItem.isChecked = !preferences.showArchived
         hideCompletedItem.isChecked = !preferences.showCompleted
+
+
+        topBarMenu = menu
+        //the habit creation menu should be disabled when numeric habits are also disabled
+        if (!preferences.isNumericalHabitsFeatureEnabled) {
```

Comment:
Since this numerical habits feature is almost complete, I think it's fine if we get rid of the feature switch. That is, let's just make the dialog available at all times. This will simplify your code quite a bit. Please feel free to delete the code to activate/deactivate this feature in the preferences screen.